### DISCUSSION TITLE
InternalQueryParser.java and InternalQueryLexer.java files

### DISCUSSION THREAD

**Discussion by Thomcray** [Epistimis/UDDL#41] [Q&A] [answered]
Finally resolved almost all the errors in the project except the InternalQueryParser.java file. It throws **Syntax error on tokens, ConstructorHeaderName expected instead** on Line **6709**, and the token is the same as that on line **6707**
![Screenshot (93)](https://user-images.githubusercontent.com/68615268/221320394-7899773f-b60f-4d24-ad6a-0de4368fe4fe.png)

Then in the InternalQueryLexer.java file, there is a Duplicate of the @SuppressWarning("all") annotation on Line **14**. I had to comment that out.
![Screenshot (94)](https://user-images.githubusercontent.com/68615268/221320813-f85dba2f-373f-4d9e-852c-d9c1b0fe60a0.png)

**Comment by steve-hickman-epistimis** [comment, marked as answer]
Those files are generated. You should not need to modify them. Any modifications you make will be lost the next time you Generate XText Artifacts. 

You should not need to manually edit any files.  

First thing you should do: Project -> Clean any time you have compile errors. This will recompile everything. This  is just to make sure any errors you see aren't residual.

If errors remain in the InternalQueryParser.java / InternalQueryLexer.java files, then you should open `Query.xtext` and Generate -> XText Artifacts to regenerate those files.

If that doesn't clean up the errors, then describe every change you've made that is not described in the instructions I've previously given.  And do this: Open a Terminal / Command Prompt window. CD into the `com.epistimis.uddl.query.parent` directory. In that directory, run the command `git diff > patch.diff` and then attach the patch.diff file you just created to this discussion so I can see what was changed.

**Reply by Thomcray** [reply]
I just regenerated the XText Artifacts for the `Query.text` file as you suggested and the error cleared, even after cleaning all projects.

Thank you for your time.

**Comment by Thomcray** [comment]
patch.diff file attached
[patch.txt](https://github.com/Epistimis/UDDL/files/10829882/patch.txt)

**Reply by steve-hickman-epistimis** [reply]
All of these are what I would expect - they just deal with classpath issues.  
1. Did you pull the most recent version of each repo?  If you forked the repos, you should make sure your forks are current.
2. Did you rebuild everything from the beginning of the Getting Started guide?

Make sure to use the [Troubleshooting Guide](https://github.com/Epistimis/UDDL/blob/master/TROUBLESHOOTING.md)

If you have done both of these things and are still having problems, then schedule time with me (1 hr) so I can look over your shoulder to see what is happening. Use this link to put something on my calendar: https://meetings.hubspot.com/steve-hickman
